dayrepo/snippets/views.py

```diff
@@ -225,117 +226,168 @@ def get(self, request, snippet_id):
-                    end_point=req.getlist("end_point")[i + 1],
-                    via_point=req.getlist("via_point")[i],
-                    client=req.getlist("client")[i],
-                    goods=req.getlist("goods")[i],
-                    load_situation=req.getlist("load_situation")[i],
-                    load_mileage=req.getlist("load_mileage")[i],
-                    hollow_mileage=req.getlist("hollow_mileage")[i],
-                    is_load_situation=req.getlist("is_load_situation")[i],
-                )
-                if process.is_load_situation == "on":
-                    process.is_load_situation = True
-                else:
-                    process.is_load_situation = False
-                post_process[i].delete()
-                process.save()
-            post_snippet=snippet
-            # トップ画面へ
-            return redirect(to="snippet_list")
+                process.is_load_situation = False
+            post_process[i].delete()
+            process.save()
+        post_snippet = snippet
+        # トップ画面へ
+        return redirect(to="snippet_list")
+
 snippet_edit = SnippetEditView.as_view()
 
+
+class post_del_pattern():
+    def snippet_del(request,target_id):
+        snippet = Snippet.objects.get(
+            pk=target_id
+        )
+        checklist = Checklist.objects.get(
+            pk=snippet.checklist_id.id
+        )
+        checklist.is_snippet_make = False
+        checklist.save()
+        dutiestrouble = DutiesTrouble.objects.get(
+            snippet_id=target_id
+        )
+        dutiestrouble.delete()
+        process = get_list_or_404(
+            Process, snippet_id=target_id
+        )
+        for process_object in process:
+            process_object.delete()
+        snippet.delete()
+
+    def checklist_del(request,target_id):
+        checklist = Checklist.objects.get(
+        pk=target_id
+        )
+        checklist.delete()
+
+    def all_del(request,target_id):
+        snippet = Snippet.objects.get(
+            pk=target_id
+        )
+        checklist = Checklist.objects.get(
+            pk=snippet.checklist_id.pk
+        )
+        dutiestrouble = DutiesTrouble.objects.get(
+            snippet_id=target_id
+        )
+        dutiestrouble.delete()
+        process = get_list_or_404(
+            Process, 
+            snippet_id=target_id
+        )
+        for process_object in process:
+            process_object.delete()
+        snippet.delete()
+        checklist.delete()
+def post_delete(request, target_id, delete_type):
+    del_class = post_del_pattern()
+    if delete_type == "snippet":
+        del_class.snippet_del(target_id)
+        return redirect(to="snippet_list")
+    elif delete_type == "checklist":
+        del_class.checklist_del(target_id)
+        return redirect(to="snippet_list")
+    elif delete_type == "all":
+        del_class.all_del(target_id)
+        return redirect(to="snippet_list")
+
+
```

Comment:
```suggestion
    if delete_type == "snippet":
        del_class.snippet_del(target_id)
    elif delete_type == "checklist":
        del_class.checklist_del(target_id)
    elif delete_type == "all":
        del_class.all_del(target_id)
        
    return redirect(to="snippet_list")
```

if-elif 文はどれかが一つ呼ばれたら他に入ることはないのと、redirect の内容が全く同じなので、1箇所で十分です

Reply:
拡張性と説明性下がらないですか？

Reply:
どういった理由でそう感じましたか？
※ここが大事

Reply:
この処理を後々、ユーザー情報や車両情報の処理にも併用したいと考えていたので、その場合、操作する画面(戻りたい画面)が異なることも考えられ、別々の方がいいかと思います。
※拡張性と説明性については、elifの一つをelseに変えるのかと思い書いてしまいました。

Reply:
> 併用したい

既に 3 ケースで同じことがわかるように、snippet_list 画面は現状 top 画面でもあるので、基本的に if 文を抜けたら snippet_list 画面になることが多いと思います.
その他の画面に飛ぶケースがあったとしても(とはいえ、 おそらくユーザー情報や車両情報を削除するときも完了後 top ページに遷移するのが普通のUXではあるので、 基本的に少ないとは思う) 、その場合でのみ return を書く方がむしろ読みやすいと感じました。

例として、仮に別のものを追加して 6 個とかになったとしましょう。

その場合以下のようになりますが、どちらの方が視認性がいいと思うかな？
※以下のコード自体は仮です。
#### 元の書き方
```
    if delete_type == "snippet":
        del_class.snippet_del(target_id)
        return redirect(to="snippet_list")
    elif delete_type == "checklist":
        del_class.checklist_del(target_id)
        return redirect(to="snippet_post")
    elif delete_type == "all":
        del_class.all_del(target_id)
        return redirect(to="snippet_list")
    elif delete_type == "user":
        del_class.all_del(target_id)
        return redirect(to="snippet_list")
    elif delete_type == "car":
        del_class.all_del(target_id)
        return redirect(to="snippet_list")
    elif delete_type == "login_user":
        del_class.all_del(target_id)
        return redirect(to="snippet_list")
```

#### 共通で一覧を用意した上で、違うところだけ明記するやり方
```
    if delete_type == "snippet":
        del_class.snippet_del(target_id)
    elif delete_type == "checklist":
        del_class.checklist_del(target_id)
        return redirect(to="snippet_post")
    elif delete_type == "all":
        del_class.all_del(target_id)
    elif delete_type == "all":
        del_class.all_del(target_id)
    elif delete_type == "all":
        del_class.all_del(target_id)
    elif delete_type == "all":
        del_class.all_del(target_id)
        
    return redirect(to="snippet_list")
```

Reply:
違うところだけ明記することは賛成ですね。
対応しました
https://github.com/bob-g12/dayrepo-for-transportation/pull/56/commits/f44997524d103e88a795f92c7be827936a6ac176